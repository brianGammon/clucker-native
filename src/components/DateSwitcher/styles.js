import { StyleSheet } from 'react-native';

const rowContainer = {
  flexDirection: 'row',
};

export default StyleSheet.create({
  rowContainer,
  dateSwitcher: {
    ...rowContainer,
    flex: 1,
    justifyContent: 'space-around',
    alignItems: 'center',
  },
  monthTitle: {
    fontWeight: 'bold',
  },
  statsLabel: {
    fontWeight: 'bold',
    fontSize: 17,
  },
  eggStats: {
    alignItems: 'center',
    paddingLeft: 10,
    paddingRight: 10,
  },
});
