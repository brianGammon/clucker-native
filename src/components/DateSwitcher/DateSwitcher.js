import React from 'react';
import moment from 'moment';
import {
  View, Button, Text, H3, H2, Icon,
} from 'native-base';
import styles from './styles';
import { type Navigation } from '../../types';

type Props = {
  mode: 'month' | 'day',
  navigation: Navigation,
  dates: {
    date: string,
    previousDate: string,
    nextDate: string,
  },
  eggCount: number | null,
};

const DateSwitcher = ({
  navigation, dates, eggCount, mode,
}: Props) => (
  <View style={styles.rowContainer}>
    <View style={styles.dateSwitcher}>
      <Button
        transparent
        dark
        onPress={() => navigation.replace(mode === 'month' ? 'Month' : 'Day', {
          date: dates.previousDate,
        })
        }
      >
        <Icon name="arrow-back" />
      </Button>
      <H2 style={styles.monthTitle}>
        {moment(dates.date).format(mode === 'month' ? 'MMMM YYYY' : 'MMMM D')}
      </H2>
      <Button
        transparent
        dark
        disabled={dates.nextDate === null}
        onPress={() => navigation.replace(mode === 'month' ? 'Month' : 'Day', {
          date: dates.nextDate,
        })
        }
      >
        <Icon name="arrow-forward" />
      </Button>
    </View>
    <View style={styles.eggStats}>
      <Text style={styles.statsLabel}>Eggs</Text>
      <H3>{eggCount || 0}</H3>
    </View>
  </View>
);

export default DateSwitcher;
