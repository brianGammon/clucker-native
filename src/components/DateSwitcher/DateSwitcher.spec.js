import React from 'react';
import DateSwitcher from '.';

describe('DateSwitcher', () => {
  const navigation = {
    navigate: () => {},
  };
  const dates = {
    date: '2018-05',
    prevDate: '20018-04',
    nextDate: '20018-06',
  };
  test('renders in month mode', () => {
    const wrapper = shallow(
      <DateSwitcher
        navigation={navigation}
        dates={dates}
        mode="month"
        eggCount={20}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('renders in day mode', () => {
    dates.date = '2018-05-05';
    dates.prevDate = '2018-05-04';
    dates.nextDate = '2018-05-06';

    const wrapper = shallow(
      <DateSwitcher
        navigation={navigation}
        dates={dates}
        mode="day"
        eggCount={20}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
