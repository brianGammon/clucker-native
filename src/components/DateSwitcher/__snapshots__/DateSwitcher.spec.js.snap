// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateSwitcher renders in day mode 1`] = `
<Styled(ViewNB)
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <Styled(ViewNB)
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-around",
      }
    }
  >
    <Styled(Button)
      dark={true}
      onPress={[Function]}
      transparent={true}
    >
      <Styled(Icon)
        name="arrow-dropleft-circle"
      />
    </Styled(Button)>
    <Styled(H3)
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      May 5, 2018
    </Styled(H3)>
    <Styled(Button)
      dark={true}
      disabled={false}
      onPress={[Function]}
      transparent={true}
    >
      <Styled(Icon)
        name="arrow-dropright-circle"
      />
    </Styled(Button)>
    <Styled(Button)
      dark={true}
      disabled={false}
      onPress={[Function]}
      transparent={true}
    >
      <Styled(Icon)
        name="today"
        type="MaterialIcons"
      />
    </Styled(Button)>
  </Styled(ViewNB)>
  <Styled(ViewNB)
    style={
      Object {
        "alignItems": "center",
        "paddingLeft": 20,
        "paddingRight": 20,
      }
    }
  >
    <Styled(Text)
      style={
        Object {
          "fontSize": 17,
          "fontWeight": "bold",
        }
      }
    >
      Eggs
    </Styled(Text)>
    <Styled(H3)>
      20
    </Styled(H3)>
  </Styled(ViewNB)>
</Styled(ViewNB)>
`;

exports[`DateSwitcher renders in month mode 1`] = `
<Styled(ViewNB)
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <Styled(ViewNB)
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-around",
      }
    }
  >
    <Styled(Button)
      dark={true}
      onPress={[Function]}
      transparent={true}
    >
      <Styled(Icon)
        name="arrow-dropleft-circle"
      />
    </Styled(Button)>
    <Styled(H3)
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      May 2018
    </Styled(H3)>
    <Styled(Button)
      dark={true}
      disabled={false}
      onPress={[Function]}
      transparent={true}
    >
      <Styled(Icon)
        name="arrow-dropright-circle"
      />
    </Styled(Button)>
    <Styled(Button)
      dark={true}
      disabled={false}
      onPress={[Function]}
      transparent={true}
    >
      <Styled(Icon)
        name="today"
        type="MaterialIcons"
      />
    </Styled(Button)>
  </Styled(ViewNB)>
  <Styled(ViewNB)
    style={
      Object {
        "alignItems": "center",
        "paddingLeft": 20,
        "paddingRight": 20,
      }
    }
  >
    <Styled(Text)
      style={
        Object {
          "fontSize": 17,
          "fontWeight": "bold",
        }
      }
    >
      Eggs
    </Styled(Text)>
    <Styled(H3)>
      20
    </Styled(H3)>
  </Styled(ViewNB)>
</Styled(ViewNB)>
`;
